source/MaterialXRuntime/Private/PvtApi.h

```diff
@@ -292,6 +307,70 @@ class PvtApi
         return _implementationSearchPaths;
     }
 
+    const FileSearchPath& getMaterialXDefinitionPaths() const
+    {
+        return _materialXDefinitionPaths;
+    }
+
+    const FileSearchPath& getGlobalDefinitionPaths() const
+    {
+        return _globalDefinitionPaths;
+    }
+
+    const FileSearchPath& getLocalDefinitionPaths() const
+    {
+        return _localDefinitionPaths;
+    }
+
+    const FileSearchPath& getMaterialXCoreDefinitionPaths() const
+    {
+        return _materialXCoreDefinitionPaths;
+    }
+
+    const FileSearchPath getDefinitionPaths(bool includeSubFolders=true) const
+    {
+        FileSearchPath definitionPaths;
+        FileSearchPath materialXDefinitionPaths = getMaterialXDefinitionPaths();
+        if (!materialXDefinitionPaths.isEmpty())
+        {
+            definitionPaths.append(materialXDefinitionPaths);
+        }
+        FileSearchPath globalDefinitionPaths = getGlobalDefinitionPaths();
+        if (!globalDefinitionPaths.isEmpty())
+        {
+            definitionPaths.append(globalDefinitionPaths);
+        }
+        FileSearchPath localDefinitionPaths = getLocalDefinitionPaths();
+        if (!localDefinitionPaths.isEmpty())
+        {
+            definitionPaths.append(localDefinitionPaths);
+        }
+        FileSearchPath materialXCoreDefinitionPaths = getMaterialXCoreDefinitionPaths();
+        if (!materialXCoreDefinitionPaths.isEmpty())
+        {
+            definitionPaths.append(materialXCoreDefinitionPaths);
+        }
+
+        if (includeSubFolders)
+        {
+            FileSearchPath childDefinitionPaths;
+            for (FileSearchPath::Iterator path = definitionPaths.begin(); path != definitionPaths.end(); ++path)
+            {
+                FilePath filePath = *path;
+                FilePathVec subdirs = filePath.getSubDirectories();
+                for (FilePath subdirPath : subdirs)
+                {
+                    childDefinitionPaths.append(subdirPath);
```

Comment:
I think we should check for duplicates here. Believe we have the same issue in loadLibraries() so would be good to only append unique paths.

Reply:
Ok, I'll add a set.